docs/sdks/server-sdks/ruby.md

```diff
@@ -19,33 +21,75 @@ gem install eppo-server-sdk
 or add to you `Gemfile`:
 
 ```
-gem 'eppo-server-sdk', '~> 0.3.0'
+gem 'eppo-server-sdk', '~> 3.0.0'
 ```
 
-## 2. Initialize the SDK
+### B. Initialize the SDK
 
-Initialize the SDK with a SDK key, which can be generated in the Eppo interface. Initialization the SDK when your application starts up to generate a singleton client instance, once per application lifecycle:
+To initialize the SDK, you will need an SDK key. You can generate one [in the flag interface](https://eppo.cloud/feature-flags/keys).
 
 ```ruby
 require 'eppo_client'
 
 config = EppoClient::Config.new('<YOUR_API_KEY>')
-client = EppoClient::init(config)
+EppoClient::init(config)
+client = EppoClient::Client.instance
 ```
 
-After initialization, the SDK begins polling Eppo’s API at regular intervals to retrieve the most recent experiment configurations such as variation values and traffic allocation. The SDK stores these configurations in memory so that assignments are effectively instant. For more information, see the [architecture overview](/sdks/overview) page.
+This generates a singleton client instance that can be reused throughout the application lifecycle.
 
-If you are using the SDK for experiment assignments, make sure to pass in an assignment logging callback (see [section](#define-an-assignment-logger-experiment-assignment-only) below).
+After initialization, the SDK begins polling Eppo's API at regular intervals to retrieve the most recent experiment configurations such as variation values and traffic allocation. The SDK stores these configurations in memory so that assignments are effectively instant. For more information, see the [architecture overview](/sdks/overview) page.
 
-:::info
+### C. Assign variations
 
-By default the Eppo client initialization is asynchronous to ensure no critical code paths are blocked. For more information on handling non-blocking initialization, see our [documentation here](/sdks/common-issues#3-not-handling-non-blocking-initialization).
+Assign users to flags or experiments using `get_<type>_assignment`, depending on the type of the flag.
+For example, for a String-valued flag, use `get_string_assignment`:
 
-:::
+```ruby
+require 'eppo_client'
+
+client = EppoClient::Client.instance
+variation = client.get_string_assignment(
+  '<FLAG-KEY>',
+  '<SUBJECT-KEY>',
+  {
+    # Optional map of subject metadata for targeting.
+  },
```

Comment:
I find `# Optional` confusing as later on you state we need to pass _something_, such as an empty dictionary. Maybe we should rephrase it to read like "Mapping of any subject metadata for targeting"

Note that Some of our other languages have syntactic sugar that let downstream developers skip it entirely, making the _parameter_ optional (as opposed to just its contents). Something we may want to do in rust/ruby land.

Lastly, in Ruby world should we say "hash" instead of "map" 🤔